cmd/antrea-agent/options.go

```diff
@@ -191,50 +198,48 @@ func (o *Options) setDefaults() {
 
 func (o *Options) validateFlowExporterConfig() error {
 	if features.DefaultFeatureGate.Enabled(features.FlowExporter) {
-		if o.config.FlowCollectorAddr == "" {
-			return fmt.Errorf("IPFIX flow collector address should be provided")
+		if strings.Contains(o.config.FlowCollectorAddr, exporter.FlowAggregatorDNSName) {
```

Comment:
I am not sure I understand why we treat this case differently?

I feel like the format should just always be `<IP / URL>[:<port>][:<proto>]` for the sake of simplicity. If everything is omitted, we default to `flow-aggregator.flow-aggregator.svc:4739:tcp`. If `<proto>` is omitted, it defaults to `tcp`. If `<port>` and `<proto>` are omitted, we default to `4739` and `tcp`.

Alternatively, `[:<port>]` and `[:<proto>]` could be swapped if you want to support service name + protocol, without having to specify the port.

Reply:
As we only open 4739 on flow aggregator service, I thought that info could be redundant.
However, I see the benefit of adding port to keep it the same for flow aggregator and external flow collector.

Let's keep the old format. Devs/users may have got used to that format. 

Instead of 4739, if we use port name defined in flow aggregator manifest, can DNS service resolve it? Any thoughts?